services/web/server/src/simcore_service_webserver/user_notifications.py

```diff
@@ -0,0 +1,86 @@
+from copy import deepcopy
+from datetime import datetime
+from enum import auto
+from typing import Any, Final
+from uuid import uuid4
+
+from models_library.users import UserID
+from models_library.utils.enums import StrAutoEnum
+from pydantic import BaseModel, NonNegativeInt, validator
+
+MAX_NOTIFICATIONS_FOR_USER: Final[NonNegativeInt] = 10
+
+
+def get_notification_key(user_id: UserID) -> str:
+    return f"user_id={user_id}"
+
+
+class NotificationCategory(StrAutoEnum):
+    NEW_ORGANIZATION = auto()
+    STUDY_SHARED = auto()
+    TEMPLATE_SHARED = auto()
+
+
+class UserNotification(BaseModel):
+    id: str
+    user_id: UserID
+    category: NotificationCategory
+    actionable_path: str
+    title: str
+    text: str
+    date: datetime
+    read: bool
+
+    def update_from(self, data: dict[str, Any]) -> None:
+        for k, v in data.items():
+            self.__setattr__(k, v)
+
+    @validator("category", pre=True)
+    @classmethod
+    def category_to_upper(cls, value: str) -> str:
+        return value.upper()
+
+    @classmethod
+    def create_from_request_data(
+        cls, request_data: dict[str, Any]
+    ) -> "UserNotification":
+        params = deepcopy(request_data)
+        params["id"] = f"{uuid4()}"
+        params["read"] = False
+        return cls.parse_obj(params)
+
+    class Config:
+        schema_extra = {
+            "examples": [
+                {
+                    "id": "123",
```

Comment:
``id``s are now uuids, in case you want to update the examples or take into account for typing or something